smart_control/dataset/dataset.py

```diff
@@ -1,96 +1,405 @@
 """Smart Buildings Dataset implementation, including loading and downloading."""
 
+from functools import cached_property
 import json
+import os
 import pickle
 import shutil
 
+from matplotlib import pyplot as plt
 import numpy as np
+import pandas as pd
 import requests
 
+from smart_control.utils.constants import ROOT_DIR
 
-class SmartBuildingsDataset:
-  """Smart Buildings Dataset"""
-
-  def __init__(self, download=True):
-    self.partitions = {
-        "sb1": [
-            "2022_a",
-            "2022_b",
-            "2023_a",
-            "2023_b",
-            "2024_a",
-        ],
-    }
-    if download:
+DATA_DIR = os.path.join(ROOT_DIR, "data")
+DOCS_DIR = os.path.join(ROOT_DIR, "docs")
+
+VALID_BUILDING_PARTITIONS = {
+    "sb1": ["2022_a", "2022_b", "2023_a", "2023_b", "2024_a"]
+}
+
+
+class BuildingDataset:
+  """A helper class for handling the dataset for a specific building.
+
+  Args:
+    building_id (str): The identifier of the building (e.g. "sb1").
```

Comment:
Here the `building_id` is "sb1" (a short code uniquely identifying the building dataset). However we see the zone information contains a different value / scheme for the `building_id` ("'buildings/3616672508'). 

In light of this, we might consider renaming the `building_id` parameter to something like `building_code` , `building_name`, or `building_dataset_id`.